xmtp_mls/src/groups/mod.rs

```diff
@@ -207,34 +207,42 @@ impl MlsGroup {
     }
 
     // Create a new group and save it to the DB
-    pub fn create_and_insert(
-        context: Arc<XmtpMlsLocalContext>,
+    pub async fn create_and_insert<ApiClient>(
+        client: &Client<ApiClient>,
         membership_state: GroupMembershipState,
         permissions: Option<PreconfiguredPolicies>,
         inbox_id: &InboxId,
-    ) -> Result<Self, GroupError> {
-        let conn = context.store.conn()?;
+    ) -> Result<Self, GroupError>
+    where
+        ApiClient: XmtpApi,
+    {
+        let conn = client.context.store.conn()?;
         let provider = XmtpOpenMlsProvider::new(conn);
         let protected_metadata = build_protected_metadata_extension(
-            &context.identity,
+            &client.context.identity,
             permissions.unwrap_or_default().to_policy_set(),
             Purpose::Conversation,
         )?;
         let mutable_metadata = build_mutable_metadata_extension_default()?;
         let group_membership = build_starting_group_membership_extension(
-            context.inbox_id(),
-            context.inbox_latest_sequence_id(),
+            client.context.inbox_id(),
+            client.inbox_latest_sequence_id().await?,
```

Comment:
What if we did this on client creation instead of on every group create? 

If it's out of date, some other client will quickly pick it up. And your client will as soon as you add a new member.

Reply:
I can add a `sequence_id` field and use that for `inbox_latest_sequence_id()`.

Reply:
> some other client will quickly pick it up

Is `other client` referring to those from the same user's device?

Reply:
Any other member of the group, including your other devices.

Reply:
Done